### DISCUSSION TITLE
Nixie and Salt Air

### DISCUSSION THREAD

**Discussion by FlaMike** [tagyoureit/nodejs-poolController#884] [Q&A] [answered]
Has anyone had any issues with their Nixie due to exposure to high humidity and salt air?  I am about 1/2 mile from the Gulf of Mexico & the humidity and salt air really take their toll on almost everything.  "Marine Grade" items begin rusting in a few years, houses & pool decks need to be power washed annually to remove algae and molds, etc.

I am considering using a conforming spray to help protect the Pi stack and all.  Curious to know if others may have had issues and if they did anything to minimize the damage to the Nixie's components.

Thanks!

**Comment by rstrouse** [comment]
Even in my box, every time I get something with a desiccant pack in it I throw that pack in the box.  Heat however, is your saving grace as only items that are cooler than the surrounding air collect condensation as a general rule.  Also, salt all by itself isn't necessarily the problem it is the galvanic corrosion that occurs as a result of moisture which makes free ions when it mixes with water and then oxidizes metals.

So the key is to keep it dry.

**Reply by FlaMike** [reply]
Thank you for your response!

I guess I need to order more things that come with desiccant packs :-)  Seriously, keeping the air inside of the box dry may be a challenge here.  In the cooler months we often get heavy fog that lasts for several days at a time.  And even in the summers, high relative humidity is typical.  I appreciate your observation that salt in and of itself is not the main culprit.  I'm checking that box off.

Wishing I coated the Pi stack boards with some conforming spray before constructing the stack & hanging the box (I'm not excited about disassembling it at this point, especially at the pad--too many itsy bitsy  parts to lose).  Nevertheless, a spritz at this point is unlikely to be damaging and may help protect  against the dreaded moisture.  I already have a can of conforming spray, so I don't have to get something that's not in my inventory.  And at least I'll be applying it outside (standing upwind and exhaling the entire time as much as possible).  For what it's worth, this is the stuff I have on hand...  Mine is an aerosol can.

https://mgchemicals.com/products/conformal-coatings/silicone-conformal-coatings/silicone-modified-conformal-coating/

As always, I appreciate your insights, help, & generosity!

**Reply by rstrouse** [reply]
Be careful with conforming spray.  It will inhibit connections and should never be sprayed inside a connector.  If there is something that needs solid connection a small and I mean small amount of dielectric grease is a better option.  However, the key is to keep moisture out so drying the air as it enters the box is a smart thing to do.  Run it through some desiccant, heck maybe even try some damp rid.  

There are a ton of pool panels out there and the electronics are not typically the corrosion problem.  Things that end up cooler than the surrounding air are the biggest issues.  As the cooler night air is heated faster than its surfaces, the moisture forms.  Most often the problem is exacerbated by improper bonding.  Galvanic corrosion on your pumps where there are dissimilar metals are a bigger target.

**Reply by FlaMike** [reply]
Good points, all.  I was thinking of using the spray once the Nixie is completed--fully wired & operational.  And only lightly coating the Pi & hats.  Hoping this approach would negate the connection challenge you cite wisely & correctly.

I understand the Pi's heat generation should minimize moisture buildup on it & possibly the entire stack.  So, perhaps this is a situation where I have a "solution" that is looking for a problem?

I also understand that, in general, pool controllers in and of themselves do not exhibit moisture/corrosion problems.  I've had both Hayward and Pentair controllers and the one problem I've not had is corrosion in the controllers.  However, none of the commercial controllers had fans moving air through them actively.  They're ventless boxes, as you know.  Although certainly not air tight, they don't add moisture through a cooling system.  With a nice, warm Pi in the mix. the situation changes a bit.

I have an Altelix 14x12x8 vented box containing a Pi 3B+, a building automation hat, 8 relay hat, Atlas Scientific Whitebox T3 hat, a EZO pH board, and a EZO ORP board.  Also connecting an Atlas Scientific pressure sensor to the T3.  Temperature for the pH calculation will be from the Pentair 10K water temperature thermister forwarded via REM.  There's also a beefy (I hope) rectifier bridge and a bunch of DIN rail blocks.  So, there's a ton of elbow room in the box, i.e., lots of air space.  All transformers & high voltage goodies remain in the ET enclosure.

I've been debating between a 120 mm 5 volt Noctura fan that spins at 1500 RPM, running 24x7 and a 120 mm 24 volt Noctura fan that spins at 3000 RPM, whenever the Pi CPU is warmer than 90 F.  I'm guessing the only time that fan wouldn't run is when freeze protection is active, but it's too soon to tell.  I don't know if the extra air movement from the 24 volt fan would help keep things cool in the summer months when the air temperature can be above 90 F for weeks at a time (yep, even at night, with plenty of humidity), or if it would exacerbate moisture introduction to the enclosure.  Any thoughts on the fan selection would be appreciated!  The 5 volt fan was already installed in the box, but the 24 volt one shows up tomorrow...

**Reply by rstrouse** [reply]
For electronics 90F is considered cold.  However, a fan in the box is to remove hot air that builds up from the CPU fan that I assume you have installed.  The Pi3B runs quite a bit cooler than the 4B but we are not processing blockchains on it either way so you should not really have any heat problems.  If you blow 100F air over a CPU consistently then it will never overheat.  The difference might be if you are blowing 200+F air over it.  Then the thermal dissipation is just not there.

My environment is one where the daytime temps reach 110 and cool off to the low 60s at night.  While summertime humidity is almost non-existent, in the winter the temps swing wildly with moist air all around.  I have never had corrosion issues with the pool but plenty of issues where there is no heat generation and metal parts.  Specifically, DMX decoders on light fixtures where the power is removed when the light is off.  Again for me this is only a wintertime thing.

**Reply by FlaMike** [reply]
I don't have a CPU fan installed at this time, just the vent fan.  I assume the primary (only?) heat generator in the enclosure to speak of is the Pi itself with a  minor contribution from the building automation hat.  According to Noctua, the 5 volt fan moves 93 cubic meters of air/hour.  The 24 volt fan, running at twice the speed, moves (surprise!) 186 cubic meters of air/hour.  Altelix says the enclosure is about 0.017 cubic meters.  If my calculations are correct, even the 5 volt fan will change over the air in the box 5470 times per hour.  Wow!

If this is even close to being correct (within an order of magnitude or 3), the 5 volt fan should keep the Pi from over heating even when blowing 90+ degree air around in the box in the heat of summer.  So, I should be good with the fan that's already in place & Mr. Bezos can get his 24 volt one back, unopened & ready to sell.

I'm guessing the Pi & its stack buddies can probably keep the interior warm enough to prevent condensation.  I'll let experience be the driver here & hold off on spraying the conforming coat unless/until a condensation problem shows up.

Whew!  I hope my reasoning makes sense.  Thanks for being such an attentive listener & provider of helpful information!!!

**Reply by rstrouse** [reply, marked as answer]
I would agree with your assessment.   You can see the nominal CPU temp from REM if you add a CPU temp device.  In the past folks have installed 3b in a Pentair enclosure in AZ sun without issue.

**Reply by FlaMike** [reply]
Thanks for guiding me through this little side trip en route to Nixie happiness!  Your patience & knowledge are phenomenal & appreciated greatly!
